{"metadata":{"symbolKind":"method","title":"makeAsyncIterator()","roleHeading":"Instance Method","role":"symbol","modules":[{"name":"FlowKit"}],"externalID":"s:7FlowKit20AsyncThrowingSubjectV04makeC8IteratorScs0G0Vyxs5Error_p_GyF","fragments":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"makeAsyncIterator"},{"text":"() -> ","kind":"text"},{"text":"AsyncThrowingStream","preciseIdentifier":"s:Scs","kind":"typeIdentifier"},{"text":"<","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Elementxmfp","text":"Element"},{"text":", ","kind":"text"},{"text":"Failure","kind":"typeIdentifier","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Failurea"},{"text":">.","kind":"text"},{"kind":"typeIdentifier","text":"Iterator","preciseIdentifier":"s:Scs8IteratorV"}]},"kind":"symbol","variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/flowkit\/asyncthrowingsubject\/makeasynciterator()"]}],"identifier":{"url":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/makeAsyncIterator()","interfaceLanguage":"swift"},"hierarchy":{"paths":[["doc:\/\/flowkit.FlowKit\/documentation\/FlowKit","doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject"]]},"abstract":[{"text":"Creates and returns an asynchronous iterator for the sequence.","type":"text"}],"schemaVersion":{"minor":3,"major":0,"patch":0},"primaryContentSections":[{"declarations":[{"languages":["swift"],"platforms":["iOS"],"tokens":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"makeAsyncIterator"},{"text":"() -> ","kind":"text"},{"text":"AsyncThrowingStream","preciseIdentifier":"s:Scs","kind":"typeIdentifier"},{"kind":"text","text":"<"},{"text":"Element","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Elementxmfp","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Failurea","text":"Failure","kind":"typeIdentifier","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/Failure"},{"text":">.","kind":"text"},{"preciseIdentifier":"s:Scs8IteratorV","text":"Iterator","kind":"typeIdentifier"}]}],"kind":"declarations"},{"content":[{"anchor":"return-value","text":"Return Value","level":2,"type":"heading"},{"inlineContent":[{"text":"An iterator that produces elements asynchronously.","type":"text"}],"type":"paragraph"}],"kind":"content"}],"sections":[],"references":{"doc://flowkit.FlowKit/documentation/FlowKit/AsyncThrowingSubject/makeAsyncIterator()":{"fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"makeAsyncIterator","kind":"identifier"},{"kind":"text","text":"() -> "},{"preciseIdentifier":"s:Scs","text":"AsyncThrowingStream","kind":"typeIdentifier"},{"text":"<","kind":"text"},{"text":"Element","kind":"typeIdentifier","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Elementxmfp"},{"text":", ","kind":"text"},{"text":"Failure","kind":"typeIdentifier","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Failurea"},{"kind":"text","text":">."},{"kind":"typeIdentifier","preciseIdentifier":"s:Scs8IteratorV","text":"Iterator"}],"role":"symbol","type":"topic","kind":"symbol","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/makeAsyncIterator()","title":"makeAsyncIterator()","abstract":[{"type":"text","text":"Creates and returns an asynchronous iterator for the sequence."}],"url":"\/documentation\/flowkit\/asyncthrowingsubject\/makeasynciterator()"},"doc://flowkit.FlowKit/documentation/FlowKit/AsyncThrowingSubject/Failure":{"type":"topic","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/Failure","url":"\/documentation\/flowkit\/asyncthrowingsubject\/failure","abstract":[{"type":"text","text":"The type of error that can terminate the sequence."}],"role":"symbol","navigatorTitle":[{"text":"Failure","kind":"identifier"}],"kind":"symbol","fragments":[{"kind":"keyword","text":"typealias"},{"kind":"text","text":" "},{"text":"Failure","kind":"identifier"}],"title":"AsyncThrowingSubject.Failure"},"doc://flowkit.FlowKit/documentation/FlowKit":{"url":"\/documentation\/flowkit","title":"FlowKit","role":"collection","abstract":[],"type":"topic","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit","kind":"symbol"},"doc://flowkit.FlowKit/documentation/FlowKit/AsyncThrowingSubject":{"type":"topic","role":"symbol","kind":"symbol","title":"AsyncThrowingSubject","url":"\/documentation\/flowkit\/asyncthrowingsubject","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject","fragments":[{"kind":"keyword","text":"struct"},{"text":" ","kind":"text"},{"kind":"identifier","text":"AsyncThrowingSubject"}],"navigatorTitle":[{"kind":"identifier","text":"AsyncThrowingSubject"}],"abstract":[{"type":"text","text":"A subject that allows for asynchronously sending values or terminating with an error."},{"text":" ","type":"text"},{"code":"AsyncThrowingSubject","type":"codeVoice"},{"text":" acts as a bridge between imperative code and asynchronous code, enabling","type":"text"},{"text":" ","type":"text"},{"text":"you to send values to an ","type":"text"},{"type":"codeVoice","code":"AsyncSequence"},{"type":"text","text":" consumer while handling potential errors gracefully."}]}}}